cpp/src/arrow/testing/matchers.h

```diff
@@ -61,6 +61,65 @@ class PointeesEqualMatcher {
 // Useful in conjunction with other googletest matchers.
 inline PointeesEqualMatcher PointeesEqual() { return {}; }
 
+class AnyOfJSONMatcher {
```

Comment:
Have tried to match this to the other matcher definitions in the file. Will this suffice?

Reply:
Thanks! I think it would technically compose a little better if it took the array instead of type + string but that's not a big deal (we can see if it's useful elsewhere first).

Reply:
(So to be clear: this is just a comment and I don't expect we should change this)